### ISSUE TITLE
Docs: Explain where to get LintResultData in order to invoke ESLint.Formatter.format()

### ISSUE THREAD

**Issue by cowwoc** [eslint/eslint#18821]
### Docs page(s)

https://eslint.org/docs/latest/integrate/nodejs-api#-loadedformatter-type

### What documentation issue do you want to solve?

https://github.com/eslint/eslint/commit/173e82040895ad53b2d9940bfb3fb67a0478f00b#diff-cb681134e1d583c87b8a91a2dfbd4de0bfe5dd84fa3ee57efb5028d7bfeb5219R632 adds a required parameter to ESLint.format() called `resultsMeta` but the example given at https://eslint.org/docs/latest/integrate/nodejs-api#-loadedformatter-type is outdated and only passes in a single parameter.

### What do you think is the correct solution?

Update the example. Show how where to get the second parameter from (does another API method return it? Is the user supposed to generate it?)

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by cowwoc** [comment]
I'm guessing this is the correct approach?

```
const resultsMeta: ESLint.LintResultData =
{
	cwd: process.cwd(),
	rulesMeta: eslint.getRulesMetaForResults(results)
};
```

**Comment by mdjermanovic** [comment]
ESLint automatically adds `cwd` and `rulesMeta` when calling the underlying loaded formatter, so there's no need to pass in these properties. In fact, they would be overwritten.

https://github.com/eslint/eslint/blob/d1f0831bac173fe3e6e81ff95c5abdbf95b02b65/lib/eslint/eslint.js#L1240-L1256

The `resultsMeta` parameter was added to `LoadedFormatter#format()` only so that the ESLint CLI can pass data related to `--max-warnings`. I think we should clarify this in the docs. If your integration doesn't have concepts like max warnings, then you don't need to pass in the second argument.

**Comment by cowwoc** [comment]
@mdjermanovic Hmm, so the index.d.ts file is wrong then. As it stands, it requires that second parameter. You should make it optional again.

**Comment by cowwoc** [comment]
I should clarify, I am invoking the Typedoc API directly from my build script (which is written in Typescript). I prefer avoiding the command-line interface if possible.

**Comment by fasttime** [comment]
> Hmm, so the distributed `index.d.ts` file is wrong then. As it stands, it requires that second parameter. You should make it optional again.

Are you talking about [this definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/23fde1f04766f759428a6302a9ee107200391429/types/eslint/index.d.ts#L1519-L1521) in DT?

```ts
    interface Formatter {
        format(results: LintResult[], data: LintResultData): string | Promise<string>;
    }
```

I guess the type of the second argument is also wrong there. Maybe a confusion with `FormatterFunction`, where the second argument is always an object:

https://github.com/eslint/eslint/blob/8824aa1469ffc572c5e5c1765d1b6da113dfba19/lib/shared/types.js#L244-L250

**Comment by cowwoc** [comment]
Yes, I was referring to that's the function I was referring to: `Formatter.format()`.
